mypy/meet.py

```diff
@@ -339,9 +339,9 @@ def _type_object_overlap(left: Type, right: Type) -> bool:
             # Or, to use a more concrete example, List[Union[A, B]] and List[Union[B, C]]
             # would be considered partially overlapping since it's possible for both lists
             # to contain only instances of B at runtime.
```

Comment:
Looking at this example I am wondering if the original behavior wasn't intentional?

Reply:
The example only has a single type argument; I think the case with multiple args was just not considered.

Reply:
I stumbled a little on this too, unions are dealt with by the recursive call, about a hundred lines earlier in the function